packages/commonwealth/client/scripts/views/components/EditProfile/EditProfile.tsx

```diff
@@ -0,0 +1,433 @@
+            label="Cancel"
+            buttonType="secondary"
+            buttonWidth="wide"
+            disabled={isUploadingProfileImage || isUploadingCoverImage}
+            onClick={() => navigate(`/profile/id/${profile.id}`)}
+          />
+          <CWButton
+            type="submit"
+            label="Save"
+            buttonType="primary"
+            buttonWidth="wide"
+            disabled={isUploadingProfileImage || isUploadingCoverImage}
+          />
+        </div>
+      </div>
+    );
+
+    return (
+      <CWPageLayout>
+        <div className="EditProfile">
+          <CWForm
+            initialValues={{
+              username: data.profile.profile_name || '',
+              email: data.profile.email || '',
+              backgroundImg: data.profile.background_image.url || '',
+              bio: deserializeDelta(data?.profile?.bio),
+            }}
+            onSubmit={handleSubmit}
+            validationSchema={editProfileValidation}
+          >
+            <div>
+              <div>
+                <CWText type="h3" fontWeight="medium">
+                  Edit Profile
+                </CWText>
+                <CWText type="b1">
+                  Add or change your general info and customize your profile.
+                </CWText>
+              </div>
+              {actionButtons}
+            </div>
+            <CWDivider />
+            <ProfileSection
+              title="General Info"
+              description="Let your community and others get to know you by sharing a bit about yourself."
+              className="input-controls"
+            >
+              <div className="profile-image-section">
+                <CWText type="caption" fontWeight="medium">
+                  Profile image
+                </CWText>
+                <CWText type="caption" className="description">
+                  Select an image from your files to upload
+                </CWText>
+                <div className="image-upload">
+                  <AvatarUpload
+                    scope="user"
+                    account={account}
+                    uploadStartedCallback={() =>
+                      setIsUploadingProfileImage(true)
+                    }
+                    uploadCompleteCallback={(files) => {
+                      setIsUploadingProfileImage(false);
+                      files.forEach((f) => {
+                        if (!f.uploadURL) return;
+                        const url = f.uploadURL.replace(/\?.*/, '').trim();
+                        setAvatarUrl(url);
+                      });
+                    }}
+                  />
+                </div>
+              </div>
+              <CWTextInput
+                fullWidth
+                placeholder="Enter your user name"
+                label="Username"
+                // TODO: unique username like in PersonalInformationStep?
+                // TODO: username generator like in PersonalInformationStep?
```

Comment:
Will have to confirm if we want to add this logic here. This is just for enhancement and not a blocker.

In case we want to add this logic, it will be in a separate PR.